code/datums/components/infective.dm

```diff
@@ -2,6 +2,9 @@
 	var/list/datum/disease/diseases //make sure these are the static, non-processing versions!
 	var/expire_time
 	var/required_clean_types = CLEAN_TYPE_DISEASE
+	/// The connect_loc_behalf component for handling movement behaviour onto a turf.
+	var/datum/component/connect_loc_behalf
```

Comment:
why are you doing this for things that dont have multiple instances of the component?

Reply:
because you cannot use GetComponent or GetExactComponent to get the connect_loc_behalf component so you need to hold a reference to it unless you want to just delete every single connect_loc_behalf component on something or do something really hacky like:
```
var/list/list_of_components = GetComponents(type)
if(length(list_of_components)
  qdel(list_of_components[1])